src-ghc/Pact/Server/PactService.hs

```diff
@@ -156,24 +153,23 @@ rollbackUpdate :: CommandEnv p -> ContMsg -> CommandState -> CommandM p ()
 rollbackUpdate CommandEnv{..} ContMsg{..} CommandState{..} = do
   -- if step doesn't have a rollback function, no error thrown. Therefore, pact will be deleted
   -- from state.
-  let newState = CommandState _csRefStore $ M.delete _cmTxId _csPacts
+  let newState = CommandState _csRefStore $ M.delete _cmPactId _csPacts
   liftIO $ logLog _ceLogger "DEBUG" $ "applyContinuation: rollbackUpdate: reaping pact "
-    ++ show _cmTxId
+    ++ show _cmPactId
   void $ liftIO $ swapMVar _ceState newState
 
-continuationUpdate :: CommandEnv p -> ContMsg -> CommandState -> CommandPact -> PactExec -> CommandM p ()
-continuationUpdate CommandEnv{..} ContMsg{..} CommandState{..} CommandPact{..} PactExec{..} = do
-  let nextStep = _cmStep + 1
-      isLast = nextStep >= _cpStepCount
+continuationUpdate :: CommandEnv p -> ContMsg -> CommandState -> PactExec -> PactExec -> CommandM p ()
```

Comment:
Having two `PactExec` side by side might lead to dev mistakes in the future. Can we `newtype` it or just `type`.

Reply:
Better yet, we don't need the old one. Fixed.